pkg/kubelet/lifecycle/handlers.go

```diff
@@ -127,8 +152,20 @@ func (hr *HandlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	path := handler.HTTPGet.Path
+	scheme := "http"
+	if string(handler.HTTPGet.Scheme) != "" {
+		scheme = string(handler.HTTPGet.Scheme)
```

Comment:
this means that any existing lifecycle handlers that specify https (and are currently running against http) will switch to https in 1.18, right? that's definitely release-note worthy, especially since it means that any that are working in 1.17 will likely break in 1.18 (assuming they were actually pointing at a http port)

Reply:
I would also expect to see an e2e test around this capability, since it appears that this is using the default http.Client, and therefore would require trusted TLS certificates.

I expected to see something similar to what the prober does around redirects and TLS trust:

https://github.com/kubernetes/kubernetes/blob/master/pkg/probe/http/http.go#L39-L59

Reply:
@liggitt Sorry for the delay here. I'm not seeing any similar e2e tests for container probes. Am I just missing those? Is it sufficient to pass an `InsecureSkipVerify: true` transport to the client running these lifecycle hooks, ala probers today?

Reply:
it looks like they are currently only tested incidentally, e.g. https://github.com/kubernetes/kubernetes/blob/84dc7046797aad80f258b6740a98e79199c8bb4d/test/e2e/apimachinery/webhook.go#L801-L805

Reply:
I am wondering if we need a feature gate to disable this function to allow a safe exit valve when/if this breaks for clusters.  I am wondering if we need to treat this like: https://github.com/kubernetes/enhancements/pull/1973
which had similar roll-out concerns.

wdyt @liggitt

Reply:
yeah, something like that seems prudent for a release